tests/test_model_builder.py

```diff
@@ -0,0 +1,294 @@
+    """
+    mb = ModelBuilder()
+    category = mb.category_component_map
+    assert len(mb.param_component_map["PX"]) == len(category["astrometry"])
+    assert len(mb.component_category_map) == len(mb.components)
+    assert len(mb.param_alias_map) == len(mb.param_component_map)
+    # test for new components
+    assert "SimpleModel" in mb.components
+    simple_comp = mb.components["SimpleModel"]
+    simple_comp.add_param(
+        floatParameter(name="TESTPARAM2", aliases=["F0"], value=0.0, unit="s")
+    )
+
+
+def test_aliases_mapping():
+    """Test if aliases gets mapped correclty
+    """
+    mb = ModelBuilder()
+    # all alases should be mapped to the components
+    assert len(mb.param_alias_map) == len(mb.param_component_map)
+
+    # Test if the param_alias_map is passed by pointer
+    _ = mb._add_alias_to_map("TESTAX", "TESTAXX", mb.param_alias_map)
+    assert "TESTAX" in mb.param_alias_map
+    # Test existing entry
+    _ = mb._add_alias_to_map("F0", "F0", mb.param_alias_map)
+    assert mb.param_alias_map["F0"] == "F0"
+    # Test repeatable_params with differnt indices.
+    for rp in mb.repeatable_param:
+        pint_par = mb.alias_to_pint_param(rp)
+        cp = mb.param_component_map[pint_par][0]
+        pint_par_obj = getattr(mb.components[cp], pint_par)
+        try:
+            prefix, id, ids = split_prefixed_name(rp)
+        except PrefixError:
+            prefix = rp
+
+        new_idx_par = prefix + "2"
+        assert mb.alias_to_pint_param(new_idx_par) == pint_par_obj.prefix + "2"
+        new_idx_par = prefix + "55"
+        assert mb.alias_to_pint_param(new_idx_par) == pint_par_obj.prefix + "55"
+        # Test all aliases
+        for als in pint_par_obj.aliases:
+            assert mb.alias_to_pint_param(als) == pint_par_obj.name
+            try:
+                als_prefix, id, ids = split_prefixed_name(als)
+            except PrefixError:
+                als_prefix = als
+        assert mb.alias_to_pint_param(als_prefix + "2") == pint_par_obj.prefix + "2"
+        assert mb.alias_to_pint_param(als_prefix + "55") == pint_par_obj.prefix + "55"
+
+
+def test_conflict_alias():
+    """Test if model builder detects the alais conflict.
+    """
+    mb = ModelBuilder()
+    # Test conflict parameter alias name
+    with pytest.raises(ConflictAliasError):
+        _ = mb._add_alias_to_map("F0", "F1", mb.param_alias_map)
+    # Define conflict alais from component class
+    class SimpleModel2(PhaseComponent):
+        """Very simple test model component
+        """
+
+        register = True
+        category = "simple_test"
+
+        def __init__(self):
+            super(SimpleModel2, self).__init__()
+            self.add_param(
+                floatParameter(name="TESTPARAMF0", aliases=["F0"], value=0.0, unit="s")
+            )
+
+    with pytest.raises(ConflictAliasError):
+        mb2 = ModelBuilder()
+    del Component.component_types["SimpleModel2"]
```

Comment:
This should be in a try/finally to be sure it gets zapped even if an exception occurs (for example if the test fails)